### PR TITLE
AArch64 JIT changes for virtual thread support

### PR DESCRIPTION
**Pull request by 0xdaryl**
* Adjust J9VMThread ownedMonitorCount counters for ARM64 monenter/monexit
* Adjust J9VMThread callOutCount counters around ARM64 direct JNI dispatch

### CODE REVIEW

**Comment by 0xdaryl** [comment]
This PR is only in draft state until PR #15552 is merged.  At that point this PR will be rebased to remove the FrontEnd changes that are needed to test it.

In the meantime, please review this PR @knn-k .  Functional testing including the VirtualThreadTest with JIT enabled succeeded.

Essentially, JNI callout counter needs to be incremented on every JNI callout that builds a frame and decremented when it returns.  For every monitor handled inline (not via a helper) the monitor count needs to incremented on monenter and decremented on monexit.

File: runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -1645,13 +1644,25 @@ J9::ARM64::TreeEvaluator::monexitEvaluator(TR::Node *node, TR::CodeGenerator *cg
...
+#if JAVA_SPEC_VERSION >= 19
+      /**
+       * Adjust J9VMThread ownedMonitorCount for execution paths that do not
+       * go out of line to release the lock.  It is safe and efficient to do
+       * this unconditionally.  There is no need to check for underflow.
+       */
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmx, node, tempReg, TR::MemoryReference::createWithDisplacement(cg, metaReg, fej9->thisThreadGetOwnedMonitorCountOffset()));
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::subimmx, node, tempReg, tempReg, 1);
+      generateMemSrc1Instruction(cg, TR::InstOpCode::strimmx, node, TR::MemoryReference::createWithDisplacement(cg, metaReg, fej9->thisThreadGetOwnedMonitorCountOffset()), tempReg);
+#endif
```
Comment (knn-k) [inline runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp:1656]: How about adding these instructions in the comment at https://github.com/eclipse-openj9/openj9/blob/2184d058bc57a01b9736dff221810dbaa5637cac/runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp#L1617-L1618 ?

File: runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp [hunk 2]
```diff
@@ -3736,6 +3758,17 @@ J9::ARM64::TreeEvaluator::monentEvaluator(TR::Node *node, TR::CodeGenerator *cg)
...
+#if JAVA_SPEC_VERSION >= 19
+      /**
+       * Adjust J9VMThread ownedMonitorCount for execution paths that do not
+       * go out of line to acquire the lock.  It is safe and efficient to do
+       * this unconditionally.  There is no need to check for overflow.
+       */
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmx, node, tempReg, TR::MemoryReference::createWithDisplacement(cg, metaReg, fej9->thisThreadGetOwnedMonitorCountOffset()));
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addimmx, node, tempReg, tempReg, 1);
+      generateMemSrc1Instruction(cg, TR::InstOpCode::strimmx, node, TR::MemoryReference::createWithDisplacement(cg, metaReg, fej9->thisThreadGetOwnedMonitorCountOffset()), tempReg);
+#endif
```
Comment (knn-k) [inline runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp:3770]: The same comment as that for `monexitEvaluator()`.
https://github.com/eclipse-openj9/openj9/blob/2184d058bc57a01b9736dff221810dbaa5637cac/runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp#L3710-L3711

Comment (0xdaryl) [reply runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp:3783]: Agreed. Fixed.

**Comment by 0xdaryl** [comment]
Jenkins test sanity alinux64 jdk17,jdk19

**Comment by 0xdaryl** [comment]
Jenkins test sanity alinux64 jdk17,jdk19

**Comment by 0xdaryl** [comment]
Jenkins test sanity amac jdk17,jdk19

**Comment by 0xdaryl** [comment]
Internal testing of all JDK 17 and 19 sanity functional, system, and openjdk testing passed on both Linux and macOS.  Un-WIPing this PR.

@knn-k : please review and merge.

**Review by knn-k — APPROVED** [review]

**Pull request merged**